.github/workflows/e2e-tests-ci-template.yml

```diff
@@ -287,7 +294,25 @@ jobs:
           cache-dependency-path: "e2e-tests/cypress/package-lock.json"
       - name: ci/e2e-test
         run: |
+          make cloud-init
+          if [ -n "$ROLLINGRELEASE_SERVER_IMAGE" ]; then
+            (
+            echo "RollingRelease: running initial smoketest against image $ROLLINGRELEASE_SERVER_IMAGE"
+            export SERVER_IMAGE="$ROLLINGRELEASE_SERVER_IMAGE"
+            export TEST_FILTER=""
+            export AUTOMATION_DASHBOARD_URL=""
+            make
+            )
+            echo "RollingRelease: asserting smoketest result has zero failures."
+            FAILURES=$(find cypress/results -name '*.json' | xargs -l jq -r '.stats.failures' | jq -s add)
+            if [ "$FAILURES" -ne "0" ]; then
+              echo "RollingRelease: initial smoketest for rolling release E2E run has nonzero ($FAILURES) failures. Aborting test run." >&2
+              exit 1
+            fi
+            rm -rfv cypress/results
+          fi
```

Comment:
Love this, on how you made it easier to reuse base setup/tests!

Reply:
Q: Is this cloud only? If yes, we should have it available for on-prem as well.

Reply:
It supports both cloud and on-prem. The `make cloud-init` and `make cloud-teardown` lines are just no-op if `SERVER=onprem`. The test run in this PR was for the onprem server, you can see that the cloud-specific lines are skipped [in the job output](https://github.com/mattermost/mattermost/actions/runs/10251879109/job/28361039268#step:5:40).

Reply:
Got it, thanks!